File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        int[] count = new int[26];
        int l = 0;
        int maxFreq = 0;
        int maxLength = 0;

        for (int r = 0; r < s.Length; r++) {
            count[s[r] - 'A']++;

            maxFreq = Math.Max(maxFreq, count[s[r] - 'A']);

            while (r - l + 1 - maxFreq > k) {
                count[s[l] - 'A']--;
                l++;
            }

            maxLength = Math.Max(maxLength, r - l + 1);
        }

        return maxLength;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)

        while l < r:
            mid = (l+r)//2

            total = sum(math.ceil(p/mid) for p in piles)

            if total <= h:
                r = mid
            else:
                l = mid + 1
        
        return l
